lib/Crypto/Encryption.php

```diff
@@ -189,7 +189,23 @@ public function begin($path, $user, $mode, array $header, array $accessList) {
 		// always use the version from the original file, also part files
 		// need to have a correct version number if they get moved over to the
 		// final location
-		$this->version = (int)$this->keyManager->getVersion($this->stripPartFileExtension($path), new View());
+		if (\OC::$server->getSession()->exists('renameFile')) {
```

Comment:
please check this location https://github.com/owncloud/core/blob/master/lib/private/Files/Storage/Wrapper/Encryption.php#L652, it is likely more suitable for the fix